src/test/tst_attrsgenerator.cpp

```diff
@@ -32,60 +32,321 @@ private slots:
+    for (int i = 0; i  < names.size(); ++i) {
+        AttributeRange* a = AttributeRange::parse(i, names.at(i), attrRges.at(i));
+        attrsScope.insert(a->attrName(), a);
+    }
+    return attrsScope;
+}
+
+void TestAttrsGenerator::_tst_parseStarCmd(QString mode)
+{
+    AttributesScope attrsScope;
+    AttrsGenerator* agen;
+    SetOfAttributes res;
+    Attributes attrs;
+    const int sizeOfAgen = 3;
+    const QStringList names = {"test0", "test1", "test2"};
+    const QStringList attrRgeStrs = {"int[0,2]", "double[2.3,7.8]", "int{-2,0,2,4,6}"};
+    const QString seed("_123");
+    QString error, cmd;
+
+    // Valid * command with empty attrScope
+    if (!strcmp(Value(mode).toString(), "rand")) {
+        cmd = QString("*%1;%2%3").arg(Value(sizeOfAgen).toQString()).arg(mode).arg(seed);
+    } else {
+        cmd = QString("*%1;%2").arg(Value(sizeOfAgen).toQString()).arg(mode);
+    }
+
+    agen = AttrsGenerator::parse(attrsScope, cmd, error);
+
+    QCOMPARE(agen->command(), cmd);
+    QCOMPARE(agen->size(), sizeOfAgen);
+
+    res = agen->create();
+
+    _tst_attrs(res, attrs, true);
+
+    // Valid * command with non-empty attrScope
+    attrsScope = _newAttrsScope(names, attrRgeStrs);
+    agen = AttrsGenerator::parse(attrsScope, cmd, error);
+
+    attrs.resize(3);
+    for (int i = 0; i < attrs.size(); ++i) {
+        attrs.replace(i, names.at(i), NULL);
+    }
+
+    QCOMPARE(agen->command(), cmd);
+    QCOMPARE(agen->size(), sizeOfAgen);
+
+    res = agen->create();
+
+    _tst_attrs(res, attrs, false);
+    _tst_mode(res, mode, attrsScope, 3);
+}
+
 void TestAttrsGenerator::tst_parseStarCmd()
 {
+    _tst_parseStarCmd("min");
+    _tst_parseStarCmd("max");
+    _tst_parseStarCmd("rand");
+}
+
+void TestAttrsGenerator::tst_parseHashCmd_min()
+{
+    AttrsGenerator* agen;
+    SetOfAttributes res;
+    Attributes attrs;
+    const int sizeOfAgen = 3;
+    const QStringList names = {"test0", "test1", "test2", "test3"};
+    const QStringList attrRgeStrs = {"int[0,2]", "double[2.3,7.8]", "int{-2,0,2,4,6}", "double{-2.2, -1.1, 0, 2.3}"};
     QString error;
 
-    // - same mode for all attributes:
-    //         '*integer;[min|max|rand_seed]'
-    /*
-    AttributesScope attrsScope =
-    QString cmd =
-    AttrsGenerator* agen = AttrsGenerator::parse(attrsScope,cmd, error);
-    */
-    /** IMPORTANT: this function is likely to be very long,
-     * if so, consider breaking into smaller functions
-     * (perhaps tst_parseStarCmd_min(), tst_parseStarCmd_max() ...) **/
+    // To shorten line:
+    const QStringList cmdList = {
+        Value(sizeOfAgen).toQString(),
+        QString("%1_min").arg(names.at(0)),
+        QString("%1_min").arg(names.at(1)),
+        QString("%1_min").arg(names.at(2)),
+        QString("%1_min").arg(names.at(3))
+    };
+    const QString cmd = QString("#%1;%2;%3;%4;%5").arg(cmdList.at(0)).arg(cmdList.at(1)).arg(cmdList.at(2)).arg(cmdList.at(3)).arg(cmdList.at(4));
```

Comment:
**pro-tip:** write as less code as possible. When the same line is written multiple times, it's usually possible to use a loop.
For example, your code below:
``` c++
const QStringList cmdList = {
    Value(sizeOfAgen).toQString(),
    QString("%1_min").arg(names.at(0)),
    QString("%1_min").arg(names.at(1)),
    QString("%1_min").arg(names.at(2)),
    QString("%1_min").arg(names.at(3))
};
const QString cmd = QString("#%1;%2;%3;%4;%5").arg(cmdList.at(0)).arg(cmdList.at(1)).arg(cmdList.at(2)).arg(cmdList.at(3)).arg(cmdList.at(4));
```
could be replaced by:
``` c++
QString cmd = QString("#%1").arg(sizeOfAgen);
for (const QString& name : names) {
    cmd += QString(";%1_min").arg(name);
}
```
which works for any number of `names`